src/Faithlife.WebRequests/Json/AutoWebServiceResponse.cs

```diff
@@ -21,13 +36,22 @@ public WebServiceException CreateException(string message = null, Exception inne
 				requestMethod: m_requestMethod,
 				requestUri: m_requestUri,
 				responseStatusCode: m_responseStatusCode,
-				responseHeaders: m_responseHeaders,
+				responseHeaders: Headers,
 				responseContentType: m_responseContentType,
 				responseContentLength: m_responseContentLength,
 				responseContentPreview: m_responseContentPreview,
 				innerException: innerException);
 		}
 
+		/// <summary>
+		/// Throws an exception if the response's content was unable to be mapped to a strongly typed property corresponding to the HTTP status code.
+		/// </summary>
+		public void ThrowIfStatusCodeNotHandled(string message = null)
```

Comment:
I'm similarly doubtful of the use-case for this method. Since it uses `IsStatusCodeHandled` it shares all my quibbles there; it also seems like I'd never call this when instead I'd call the existing `VerifyResultIsExpected` extensions.

It's also part of a breaking change to clients that *don't* want to take advantage of this functionality, in that they expect this exception to be thrown without intervention on their behalf.

Combined with Ed's suggestion of only populating the `StatusCode` property if it's provided by the derived object, we can solve the "breaking change" issue by reverting to the old "throw exception if `!isStatusCodeHandled`" logic when there is no `StatusCode` property provided.